diff -rupN gtk-0.14.3/Graphics/UI/Gtk/Embedding/Types.chs new-gtk-0.14.3/Graphics/UI/Gtk/Embedding/Types.chs
--- gtk-0.14.3/Graphics/UI/Gtk/Embedding/Types.chs	2016-05-22 03:45:58.000000000 +0200
+++ new-gtk-0.14.3/Graphics/UI/Gtk/Embedding/Types.chs	2025-08-01 21:34:55.810556987 +0200
@@ -45,7 +45,7 @@ import Foreign.ForeignPtr (ForeignPtr, c
 #if __GLASGOW_HASKELL__ >= 707
 import Foreign.ForeignPtr.Unsafe (unsafeForeignPtrToPtr)
 #else
-import Foreign.ForeignPtr (unsafeForeignPtrToPtr)
+import Foreign.ForeignPtr.Unsafe (unsafeForeignPtrToPtr)
 #endif
 import Foreign.C.Types    (CULong(..), CUInt(..), CULLong(..))
 import System.Glib.GType  (GType, typeInstanceIsA)
diff -rupN gtk-0.14.3/Graphics/UI/Gtk/Types.chs new-gtk-0.14.3/Graphics/UI/Gtk/Types.chs
--- gtk-0.14.3/Graphics/UI/Gtk/Types.chs	2016-05-22 03:45:58.000000000 +0200
+++ new-gtk-0.14.3/Graphics/UI/Gtk/Types.chs	2025-08-01 21:35:08.841037917 +0200
@@ -772,7 +772,7 @@ import Foreign.ForeignPtr (ForeignPtr, c
 #if __GLASGOW_HASKELL__ >= 707
 import Foreign.ForeignPtr.Unsafe (unsafeForeignPtrToPtr)
 #else
-import Foreign.ForeignPtr (unsafeForeignPtrToPtr)
+import Foreign.ForeignPtr.Unsafe (unsafeForeignPtrToPtr)
 #endif
 import Foreign.C.Types    (CULong(..), CUInt(..), CULLong(..))
 import System.Glib.GType  (GType, typeInstanceIsA)
